src/lightgbm_R.cpp

```diff
@@ -426,19 +426,20 @@ LGBM_SE LGBM_BoosterGetEvalNames_R(LGBM_SE handle,
   LGBM_SE buf_len,
   LGBM_SE actual_len,
   LGBM_SE eval_names,
-  LGBM_SE call_state) {
+  LGBM_SE call_state,
+  LGBM_SE data_idx) {
```

Comment:
this will break R package, you should update the R files accordingly.
BTW, the call state should be in the last argument.